packages/common/src/types/filterGrammarConversion.ts

```diff
@@ -0,0 +1,130 @@
+    isDateFilterRule,
+    type DateFilterSettings,
+    type MetricFilterRule,
+} from './filter';
+import filterGrammar from './filterGrammar';
+
+const isValidMetricFilterRule = (filter: MetricFilterRule): boolean => {
+    const { values, operator } = filter;
+    const operatorAllowsEmptyValues = [
+        FilterOperator.NULL,
+        FilterOperator.NOT_NULL,
+    ].includes(operator);
+    const firstValue = values?.[0] as string | number | undefined;
+    return operatorAllowsEmptyValues || firstValue !== undefined;
+};
+/**
+ * Converts a filter rule from custom metrics to dbt meta tags in yml
+ * More info about dbt filters: https://docs.lightdash.com/references/metrics/#available-filter-types
+ */
+const convertFilterOperatorToDbt = (filter: MetricFilterRule): string[] => {
+    const { values, operator } = filter;
+
+    if (!isValidMetricFilterRule(filter)) {
+        throw new ParameterError(
+            `Filter values are undefined for filter: ${JSON.stringify(filter)}`,
+        );
+    }
+    // Operators without values are handled here
+    switch (operator) {
+        case FilterOperator.NULL:
+            return [`null`];
+        case FilterOperator.NOT_NULL:
+            return [`!null`];
+        default:
+            break; // continue
+    }
+    const validValues = values as (string | number)[];
+    const firstValue = validValues[0];
+
+    switch (operator) {
+        case FilterOperator.EQUALS:
+            return validValues?.map((value) => `${value}`);
+        case FilterOperator.NOT_EQUALS:
+            return [`!${firstValue}`];
+        case FilterOperator.INCLUDE:
+            return [`%${firstValue}%`];
+        case FilterOperator.NOT_INCLUDE:
+            return [`!%${firstValue}%`];
+        case FilterOperator.STARTS_WITH:
+            return [`${firstValue}%`];
+        case FilterOperator.ENDS_WITH:
+            return [`%${firstValue}`];
+        case FilterOperator.GREATER_THAN:
+            return [`> ${firstValue}`];
+        case FilterOperator.GREATER_THAN_OR_EQUAL:
+            return [`>= ${firstValue}`];
+        case FilterOperator.LESS_THAN:
+            return [`< ${firstValue}`];
+        case FilterOperator.LESS_THAN_OR_EQUAL:
+            return [`<= ${firstValue}`];
+        case FilterOperator.IN_THE_NEXT:
+        case FilterOperator.IN_THE_PAST:
+            if (isDateFilterRule(filter)) {
+                const settings = filter.settings as DateFilterSettings;
+                const unitOfTime = settings.unitOfTime || UnitOfTime.days;
+
+                let totalValue = Number(firstValue);
+                if (
+                    settings.completed &&
+                    operator === FilterOperator.IN_THE_NEXT
+                ) {
+                    totalValue += 1;
+                }
+                if (
+                    settings.completed &&
+                    operator === FilterOperator.IN_THE_PAST
+                ) {
+                    totalValue -= 1;
```

Comment:
I would throw `NotImplementedError` when complete=true rather than returning the wrong filter logic.
Filter in UI
`date is in the last 1 completed year` (2024.01.01 - 2024.12.31)
converted to YML (1-1=0)
`date is in the last 0 year` 🤔